### DISCUSSION TITLE
child component args

### DISCUSSION THREAD

**Discussion by lamtech58** [arkivanov/Decompose#590] [Q&A] [answered]
Why my app start over from Splash screen when i put it on background with the following child component arguments : 

```
is RootComponent.Configuration.ProductDetailsScreen -> RootComponent.Child.ProductDetailsScreen(
                ProductDetailsComponent(
                    componentContext,
                    shoppingBagUDataSource = shoppingBagUDataSource,
                    product = configuration.product,
                    productReference = configuration.productReference,
                    onOutputEvent = onOutputEvent,
                    onNavigationEvent = ::sendNavigationEvent
                )
            )
```

but without the product and productReference , it resumes where i left it in the ProductDetailsUi , the following code is without those args :

```
is RootComponent.Configuration.ProductDetailsScreen -> RootComponent.Child.ProductDetailsScreen(
                ProductDetailsComponent(
                    componentContext,
                    shoppingBagUDataSource = shoppingBagUDataSource,
                    product = null, //configuration.product,
                    productReference = null, // configuration.productReference,
                    onOutputEvent = onOutputEvent,
                    onNavigationEvent = ::sendNavigationEvent
                )
            )
```

**Comment by arkivanov** [comment]
It's hard to say based on the provided code. I suggest to debug with logs, most likely somewhere the navigation is being performed to the splash screen. A runnable reproducer project would be also helpful.

**Reply by lamtech58** [reply, marked as answer]
Found out that it's firebase timestamp serialization issue

**Reply by arkivanov** [reply]
Thanks for sharing!

**Reply by lamtech58** [reply]
You're welcome, can you please provide a link for better kotlinx custom serialization?

**Reply by arkivanov** [reply]
I really suggest Chapters 3 and 4 in the [official docs](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serialization-guide.md).

**Reply by lamtech58** [reply]
Thanks a lot
